test/extensions/filters/http/dynamic_forward_proxy/proxy_filter_integration_test.cc

```diff
@@ -15,12 +15,15 @@ namespace {
 class ProxyFilterIntegrationTest : public testing::TestWithParam<Network::Address::IpVersion>,
                                    public HttpIntegrationTest {
 public:
-  ProxyFilterIntegrationTest() : HttpIntegrationTest(Http::CodecType::HTTP1, GetParam()) {}
+  ProxyFilterIntegrationTest() : HttpIntegrationTest(Http::CodecType::HTTP1, GetParam()) {
+    upstream_tls_ = true;
+  }
 
   void initializeWithArgs(uint64_t max_hosts = 1024, uint32_t max_pending_requests = 1024,
                           const std::string& override_auto_sni_header = "") {
     setUpstreamProtocol(Http::CodecType::HTTP1);
     const std::string filename = TestEnvironment::temporaryPath("dns_cache.txt");
+    ::unlink(filename.c_str());
```

Comment:
Out of curiosity, with your new O_TRUNC writes, is this still required?

Reply:
I think so - not all the tests write the file and the file shouldnt exist in the tests which don't use one